unit_tests/test_redis_aggregator.py

```diff
@@ -125,25 +99,24 @@ def test_constructor_pubsub_instantiated(aggregator):
     def test_constructor_client_command_buffer_instantiated():
         client_command_buffer_mock = MagicMock()
         with patch("gsy_e_sdk.redis_aggregator.ClientCommandBuffer",
-                   return_value=client_command_buffer_mock) as mocked_class:
+                   return_value=client_command_buffer_mock):
             agg = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
-            mocked_class.assert_called()
         assert agg._client_command_buffer is client_command_buffer_mock
 
     @staticmethod
     @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
     def test_constructor_threadpoolexecutor_instantiated():
-        ThreadPoolExecutor_mock = MagicMock
+        thread_pool_executor_mock = MagicMock
         with patch("gsy_e_sdk.redis_aggregator.ThreadPoolExecutor",
-                   return_value=ThreadPoolExecutor_mock) as mocked_class:
+                   return_value=thread_pool_executor_mock) as mocked_class:
             agg = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
             mocked_class.assert_called()
```

Comment:
Nit: please not that there the same as https://github.com/gridsingularity/gsy-e-sdk/pull/197#discussion_r775449592 applies. This can also be applied to other parts of the PR.

Reply:
I see this. I've been thinking however: Would you say this also applies to the `assert_called_with(_some_args_)` expression?

Reply:
I think this depends on the type of message. 
- If we're talking about an outgoing query, then we don't really care about what arguments we pass. That's because we can just ignore the outgoing query message, without mocking them or setting expectations. As we've seen in the past, we just use their outputted results.
- If we're talking about an outgoing command, then we probably care about the parameters. That's because we will not check the side-effects of outgoing commands, but we will just create a mock and expect to be sending the right content to the other class/method.
- If we're talking about private commands (sent to self), then we usually don't mock/stub/patch them. We just check their side-effects inside of our instance.

Also, please note the difference between "biding to the implementation" and "testing the interface". If an interface (public method) accepts certain parameters, then it's correct to check that we're passing those parameters. We mostly talk about "implementation" when we're referring to methods that are not part of the public interface. It might not be always the case, but it might help.